124-exec-engine.md

```diff
@@ -0,0 +1,253 @@
+---
+title: About `exec` engine
+editor: visual 
+---
+
+# How does this engine works ?
+
+The `exec` engine allows to execute an arbitrary command on the code chunk content, optionally with arguments. If any, they are passed to `engine.opts` and they could be one of the following:
+
+``` r
+engine.opts = list(command, file, ext, clean, args, args1, args2)
+```
+
+The execution of the chunk will be equivalent to this command line
+
+    <command> <args1> <args> <args2>
+
+Behavior by default, during evaluation will be:
+
+-   Retrieve the path of the command in `command` option.
+-   Write the chunk content into a temporary file `f`
+-   Register the `clean` function to execute on exit, basically to remove the temp file `f`.
+-   Compute the argument of the command line by appending the results of `args1()`, with `args()` then `args2()`.
+
+There should be very rare usage of changing the default behavior, but it can be useful to debug for example or other advanced usage.
+
+You can find some examples below on how to use each option. We are using `Rscript` an d `python` only in the examples below by convenience but this work with any executable file.
+
+# Simple example
+
+`command` can be the name of a command in PATH.
+
+
+```r
+1 + 1
+```
+
+```
+## [1] 2
+```
+
+
+```python
+print(1 + 1)
+```
+
+```
+## 2
+```
+
+or a full path
+
+
+```python.exe
```

Comment:
I get this on Windows when passing a full path because of the extension. We may want to use `xfun::sans_ext()` to be sure to remove it ?

Reply:
Yes, and done.